auth/access_token_from_impersonated_credentials.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START auth_cloud_accesstoken_impersonated_credentials]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
+)
+
+// getAccessTokenFromImpersonatedCredentials uses a service account (SA1) to impersonate
+// another service account (SA2) and obtain OAuth2 token for the impersonated account.
+// To obtain a token for SA2, SA1 should have the "roles/iam.serviceAccountTokenCreator" permission on SA2.
+func getAccessTokenFromImpersonatedCredentials(w io.Writer, impersonatedServiceAccount, scope string) error {
+	// impersonatedServiceAccount := "name@project.service.gserviceaccount.com"
+	// scope := "https://www.googleapis.com/auth/cloud-platform"
```

Comment:
Prefer to minimize the use of parameters to maximize readability of the code.
https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments

suggestion: Perhaps we should hard-code the scope?

Reply:
Is it ok to leave the scope parameter in to maintain parity with the other languages? 

See:

- https://github.com/GoogleCloudPlatform/python-docs-samples/pull/10759/files#diff-b153b82f025667cd97795ed13bca0d482ae15be712050e6a1ec5c202a6cfc7eb
- https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3541/files#diff-912d17734164b6b68691f390f549c97e30ae27b33022bcde11c855a599be5ac1
- https://github.com/GoogleCloudPlatform/java-docs-samples/pull/8747/files#diff-331f807619d1389073d70d053199d21a174b410ed5e21f6ca2d9c1978956f670

Reply:
Given it's already in, this is ok, but it would be worth considering making the suggested change to all languages.

Reply:
ok, thanks!